dash/include/dash/memory/GlobStaticMem.h

```diff
@@ -294,6 +288,23 @@ class GlobStaticMem
     return _nlelem;
   }
 
+  index_type distance(const pointer & gbegin, const pointer & gend) const {
```

Comment:
This "stuff" is the calculation of global pointer distance which depends on `Glob*Mem` as discussed in #333 so it cannot be implemented as a single independent function.
For comparison, see `dash::GlobUnitMem`.

We could specialize `dash::distance` for `GlobPtr` and its dependent `Glob*Mem` and define `GlobPtr::operator-(const self_t &)` as `dash::distance(*this, other)`.